sstables/sstables.cc

```diff
@@ -550,20 +550,20 @@ future<> parse(sstable_version_types v, random_access_reader& in, summary& s) {
                 s.positions.push_back(s.header.memory_size);
                 return make_ready_future<>();
             });
-        }).then([v, &in, &s] {
+        }).then([v, schema, &in, &s] {
             in.seek(sizeof(summary::header) + s.header.memory_size);
-            return parse(v, in, s.first_key, s.last_key);
-        }).then([&in, &s] {
+            return parse(schema, v, in, s.first_key, s.last_key);
+        }).then([schema, &in, &s] {
             in.seek(s.positions[0] + sizeof(summary::header));
             s.entries.reserve(s.header.size);
 
-            return do_with(int(0), [&in, &s] (int& idx) mutable {
-                return do_until([&s] { return s.entries.size() == s.header.size; }, [&s, &in, &idx] () mutable {
+            return do_with(int(0), [schema, &in, &s] (int& idx) mutable {
```

Comment:
some of these schema captures, like this one, can probably be done with move. Not all of them, however :-(